osu.Game/Screens/OsuScreen.cs

```diff
@@ -103,8 +109,18 @@ private void load(OsuGameBase game, OsuGame osuGame, AudioManager audio)
             if (osuGame != null)
             {
                 Ruleset.BindTo(osuGame.Ruleset);
-                hideOverlaysOnEnter.BindTo(osuGame.HideOverlaysOnEnter);
-                allowOpeningOverlays.BindTo(osuGame.AllowOpeningOverlays);
+                AllowOverlays.BindTo(osuGame.AllowOverlays);
```

Comment:
This is seriously dangerous. You're both binding to the global bindable and changing the value of it in BDL async methods. This means if a screen is preloaded without being immediately pushed, it is potentially changing the state of this variable.